homeassistant/components/brunt/cover.py

```diff
@@ -2,84 +2,99 @@
+    DOMAIN,
+    FAST_INTERVAL,
+    OPEN_POSITION,
+    REGULAR_INTERVAL,
+)
 
 _LOGGER = logging.getLogger(__name__)
 
 COVER_FEATURES = SUPPORT_OPEN | SUPPORT_CLOSE | SUPPORT_SET_POSITION
 
-ATTR_REQUEST_POSITION = "request_position"
-NOTIFICATION_ID = "brunt_notification"
-NOTIFICATION_TITLE = "Brunt Cover Setup"
-ATTRIBUTION = "Based on an unofficial Brunt SDK."
-
-CLOSED_POSITION = 0
-OPEN_POSITION = 100
-
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {vol.Required(CONF_USERNAME): cv.string, vol.Required(CONF_PASSWORD): cv.string}
-)
 
-
-def setup_platform(hass, config, add_entities, discovery_info=None):
+async def async_setup_entry(
+    hass, entry, async_add_entities, discovery_info=None
+) -> None:
     """Set up the brunt platform."""
+    bapi: BruntClientAsync = hass.data[DOMAIN][entry.entry_id]["bapi"]
+    coordinator: DataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id][
+        "coordinator"
+    ]
+    async_add_entities(
+        BruntEntity(coordinator, serial, thing, bapi, entry.entry_id)
+        for serial, thing in coordinator.data.items()
+    )
 
-    username = config[CONF_USERNAME]
-    password = config[CONF_PASSWORD]
-
-    bapi = BruntAPI(username=username, password=password)
-    try:
-        things = bapi.getThings()["things"]
-        if not things:
-            _LOGGER.error("No things present in account")
-        else:
-            add_entities(
-                [
-                    BruntDevice(bapi, thing["NAME"], thing["thingUri"])
-                    for thing in things
-                ],
-                True,
-            )
-    except (TypeError, KeyError, NameError, ValueError) as ex:
-        _LOGGER.error("%s", ex)
-        hass.components.persistent_notification.create(
-            "Error: {ex}<br />You will need to restart hass after fixing.",
-            title=NOTIFICATION_TITLE,
-            notification_id=NOTIFICATION_ID,
-        )
-
-
-class BruntDevice(CoverEntity):
-    """
-    Representation of a Brunt cover device.
 
-    Contains the common logic for all Brunt devices.
-    """
+class BruntEntity(CoordinatorEntity, CoverEntity):
+    """Representation of a Brunt cover entity."""
 
     _attr_device_class = DEVICE_CLASS_WINDOW
     _attr_supported_features = COVER_FEATURES
 
-    def __init__(self, bapi, name, thing_uri):
-        """Init the Brunt device."""
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        serial: str,
+        thing: Thing,
+        bapi: BruntClientAsync,
+        entry_id: str,
+    ) -> None:
+        """Init the Brunt entity."""
+        super().__init__(coordinator)
         self._bapi = bapi
-        self._attr_name = name
-        self._thing_uri = thing_uri
+        self._thing = thing
+        self._entry_id = entry_id
+
+        self._attr_unique_id: str = serial
+        self._attr_name: str = self._thing.NAME
+        self._attr_extra_state_attributes = {
+            ATTR_ATTRIBUTION: ATTRIBUTION,
```

Comment:
Attribution is set by returning it from that entity property nowadays.

https://developers.home-assistant.io/docs/core/entity#generic-properties